### PR TITLE
Handle looping edge cases

### PR DESCRIPTION
**Pull request by devmanhinton**
@bglusman Yo, yo, yo. EndDash looping now handles edgeCases.

Currently that just means a div in data-each that is unnamed

``` html
<div 'kindPolymorphic-' data-each>
  <div class='whenUser-'>
  </div>
  <div data-im-base-case*>
  </div>
</div>
```

*this is not necessary just for fun!

We can decided on a reserved word if we want. That is easy to do.

### CODE REVIEW

File: lib/reactions/looping.js [hunk 1]
```diff
@@ -12,12 +12,11 @@ var LoopingReaction = Reaction.extend({
   },
 
   getTemplateClass: function(model) {
-    if (this.polymorphicKey) {
-      var value = get(model, this.polymorphicKey);
-      return this.itemTemplates[value];
-    } else {
-      return this.itemTemplates["default"];
-    }
+      var template = this.itemTemplates[get(model, this.polymorphicKey) || "default"];
```
Comment (bglusman) [inline lib/reactions/looping.js]: looks like this got indented one too far

Comment (devmanhinton) [inline lib/reactions/looping.js]: You are right. Thank you catching. It is fixed.

File: lib/reactions/looping.js [hunk 2]
```diff
@@ -12,12 +12,11 @@ var LoopingReaction = Reaction.extend({
...
+      if (!template) {
+        template = this.itemTemplates["default"];
```
Comment (bglusman) [inline lib/reactions/looping.js]: It's a little confusing why there's the || "default" above and then THIS 'default' here....  shouldn't line 15 be doing exactly what this line is doing, if there is a "default" template, without this line being there?

**Comment by devmanhinton** [comment]
@bglusman It should be clear now. It is a bit tricky code.

**Pull request merged**